server/routes/incidentDetails/incidentDetails.ts

```diff
@@ -42,40 +49,92 @@ export default class IncidentDetailsRoutes {
+      associatedPersonDetails.name = associatedPrisoner.displayName
     }
 
     return res.render(`pages/incidentDetails`, {
       errors: error ? [error] : [],
+      prisonerNumber,
       prisoner,
       locations,
       data,
       reportingOfficer: reporter || '',
       submitButtonText: 'Save and continue',
       reportPreviouslySubmitted: false,
+      queryRadioSelection,
+      inciteAnotherPrisonerInput,
+      assistAnotherPrisonerInput,
+      associatedPersonDetails,
     })
   }
 
   view = async (req: Request, res: Response): Promise<void> => this.renderView(req, res, {})
 
   submit = async (req: Request, res: Response): Promise<void> => {
-    const { incidentDate, locationId } = req.body
+    const {
+      incidentDate,
+      locationId,
+      currentRadioSelected,
+      search,
+      inciteAnotherPrisonerInput,
+      assistAnotherPrisonerInput,
+    } = req.body
     const { user } = res.locals
     const { prisonerNumber } = req.params
+    const selectedPerson = JSON.stringify(req.query.selectedPerson)?.replace(/"/g, '')
 
-    const error = validateForm({ incidentDate, locationId })
+    if (search) {
+      const searchValue =
+        search === 'inciteAnotherPrisonerSearchSubmit' ? inciteAnotherPrisonerInput : assistAnotherPrisonerInput
+      const error = validatePrisonerSearch({ searchTerm: searchValue, inputId: currentRadioSelected })
+      if (error)
+        return this.renderView(req, res, {
+          error,
+          incidentDate,
+          locationId,
+          queryRadioSelection: currentRadioSelected,
+          inciteAnotherPrisonerInput,
+          assistAnotherPrisonerInput,
+        })
+      req.session.redirectUrl = `/incident-details/${prisonerNumber}?queryRadioSelection=${currentRadioSelected}`
+      req.session.incidentDate = incidentDate
+      req.session.incidentLocation = locationId
+      const searchPageHref = `/select-associated-prisoner/${prisonerNumber}?searchTerm=${searchValue}`
+      return res.redirect(`${searchPageHref}&startUrl=incident-details`)
+    }
+
+    // Check the selectedPerson has not been tampered with in URL
+    const associatedPrisonersNumber = isPrisonerIdentifier(selectedPerson) ? selectedPerson : null
+
+    const error = validateForm({
+      incidentDate,
+      locationId,
+      incidentRole: currentRadioSelected,
+      associatedPrisonersNumber,
+    })
     if (error) return this.renderView(req, res, { error, incidentDate, locationId })
 
     try {
       const newAdjudication = await this.placeOnReportService.startNewDraftAdjudication(
         formatDate(incidentDate),
         locationId,
         prisonerNumber,
+        associatedPrisonersNumber,
+        currentRadioSelected, // Somewhere before(?) this we need to convert the currentRadioSelected to the calculated roleCode
         user
       )
+      delete req.session.redirectUrl
```

Comment:
Delete as part of the get(), which is where it should be read in answering another PR